tests/neg/OrType.scala

```diff
@@ -2,7 +2,7 @@ class B(val x: Int)
 class C(val x: Double)
 
 object Test{
-  def bar(x: B | C): Int | Double = x.x
+  def bar(x: B | C): Int | Double = x.x  // error
```

Comment:
I assume this is caused by the change to `findMember`, we now always approximate union prefixes, I think this should be reconsidered since it considerably diminishes the usability of union types, it also breaks the nice symmetry with selection on intersection types.